.gitlab-ci-check-golang-unittests-v2.yml

```diff
@@ -0,0 +1,113 @@
+# .gitlab-ci-check-golang-unittests.yml
+#
+# This gitlab-ci template runs unit tests and publishes code coverage
+# from a Go repository
+#
+# Add it to the project in hand through Gitlab's include functionality
+#
+# include:
+#   - project: 'Northern.tech/Mender/mendertesting'
+#     file: '.gitlab-ci-check-golang-unittests.yml'
+#
+# If the repo has some compile time dependencies, the template expect to
+# exist a deb-requirements.txt file with the Debian OS required packages
+#
+# The template uses latest Mender supported golang version.
+# To override the version to use, append to your .gitlab-ci.yml:
+# test:unit:
+#   image: golang:1.11.4
+#
+# Requires the following variables set in the project CI/CD settings:
+#   COVERALLS_TOKEN: Token from coveralls.io for this repository
+#
+# By default, it installs MongoDB version 4.4 for Debian buster. You can
+# specify the Debian release name for the upstream APT repository and
+# the MongoDB version overriding the variables below:
+#
+# test:unit:
+#   variables:
+#     MONGODB_DEBIAN_RELEASE: "stretch"
+#     MONGODB_VERSION: "3.6"
+#
+
+stages:
+  - test
+  - publish
+
+test:unit:
+  tags:
+    - mender-qa-worker-generic-light
+  stage: test
+  needs: []
+  except:
+    - /^saas-[a-zA-Z0-9.]+$/
+  image: golang:1.20-alpine3-17
+  services:
+    - mongo:4.4
+  variables:
+    GIT_STRATEGY: clone # clone entire repo instead of reusing workspace
+    GIT_DEPTH: 0 # avoid shallow clone, this test requires full git history
+    TEST_MONGO_URL: "mongodb://mongo"
+  before_script:
+    # Install JUnit test reporting formatter
+    - go install github.com/jstemmer/go-junit-report@v1.0.0
+
+  script:
+    - >-
+      go test ./... -v -covermode=atomic -coverprofile=coverage.txt 2>&1 |
```

Comment:
Does this collect the coverage of all packages? Why then are we doing it [this complicated way](https://github.com/mendersoftware/mendertesting/blob/9ed85cd33d5ead852b3b7fd7b589b95debd854d6/.gitlab-ci-check-golang-unittests.yml#L86) in the old template?

Reply:
Yes it does, but it omits packages that don't have `_test.go` files, which I believe is what we want anyway. I have no idea why we run go test for each individual package, to me it just looks like unnecessary boilerplate, but maybe it has something to do with how coveralls collects coverage?

Reply:
The run for each package is older than coveralls so it cannot be coverage service specific... I also see it as unnecessary complex way of running the tests, but TBH never really questioned it :sweat_smile: 

I think it is safe to merge as-is: this is a new template so we will need to explicitly include it in the child repos with a PR, and then we could potentially see if there is anything missing (a drop in the coverage).